src/sidebar/components/new-note-btn.js

```diff
@@ -1,21 +1,44 @@
 'use strict';
 
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+
 const events = require('../events');
+const useStore = require('../store/use-store');
+const { applyTheme } = require('../util/theme');
+const { withServices } = require('../util/service-context');
 
-module.exports = {
-  controllerAs: 'vm',
-  //@ngInject
-  controller: function($rootScope, store) {
-    this.onNewNoteBtnClick = function() {
-      const topLevelFrame = store.frames().find(f => !f.id);
-      const annot = {
-        target: [],
-        uri: topLevelFrame.uri,
-      };
+function NewNoteBnt({ $rootScope, settings }) {
+  const store = useStore(store => ({
+    createAnnotation: store.createAnnotation,
+    frames: store.frames,
+  }));
```

Comment:
In this case, correct me if I'm wrong, but because the frames() gets executed on the click, then in theory, it will gather the state at that time and provide it to the click handler. But of course, if this was something that rendered the output to the DOM directly, then we would not capture live store changes as intended.

A few things that come to mind. I think we do need a really clear standard for selectors vs actions since we don't have `mapStateToProps ` or `mapDispatchToProps`. We simply have a single method that sorta facilitates both.

perhaps `useStore` vs `useActions`

For example, from the `useStore` callback ...

`myFrames: store.frames()`
vs
`myFrames: store.getState().frames`

Both do the same thing. Are we discouraged form using the latter? What if we want to access a store, but we have no selector to get that store. Should every store have a getter selector? I think they also should all be name standardized. `frames()` vs `getGroups()`. I'm just thinking out-loud here :)

Reply:
I agree and I’ve had the question myself. Personally what I’ve been standardizing to in my own code is to use the getState() since I feel it’s redundant to have a better function as well.